File: backup/src/jclang/structs/CXIdxEntityInfo.java


package jclang.structs;

import com.sun.jna.Structure;
import com.sun.jna.ptr.PointerByReference;
import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
@Structure.FieldOrder({"kind", "templateKind", "lang", "name", "USR", "cursor", "attributes", "numAttributes"})
public class CXIdxEntityInfo extends Structure {
    public int kind;
    public int templateKind;
    public int lang;
    public String name;
    public String USR;
    public CXCursor.ByValue cursor;
    @Nullable
    public PointerByReference /* CXIdxAttrInfo */ attributes;
    public int numAttributes;

    public static class ByReference extends CXIdxEntityInfo implements Structure.ByReference {}
}

File: klang/libclang/src/main/kotlin/klang/jvm/binding/CXIdxEntityInfo.kt
package klang.jvm.binding

import com.sun.jna.Structure
import com.sun.jna.ptr.PointerByReference
import klang.jvm.binding.CXCursor.CXCursorByValue

@Suppress("unused")
@Structure.FieldOrder("kind", "templateKind", "lang", "name", "USR", "cursor", "attributes", "numAttributes")
open class CXIdxEntityInfo : Structure() {
	@JvmField
    var kind = 0
	@JvmField
    var templateKind = 0
	@JvmField
    var lang = 0
	@JvmField
    var name: String = ""
	@JvmField
    var USR: String = ""
	@JvmField
    var cursor: CXCursorByValue = CXCursorByValue()
	@JvmField
    var  /* CXIdxAttrInfo */attributes: PointerByReference = PointerByReference()
	@JvmField
    var numAttributes = 0

    class ByReference : CXIdxEntityInfo(), Structure.ByReference
}
